### PR TITLE
Add Gazebo builtin plugins to LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by chapulina**
Should fix some failing tests. Addresses #970

### CODE REVIEW

File: gazebo_plugins/env-hooks/gazebo_plugins.sh.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# generated from gazebo_plugins/env-hooks/gazebo_plugins.sh.in
+
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:@GAZEBO_PLUGIN_PATH@
```
Comment (wjwwood) [inline gazebo_plugins/env-hooks/gazebo_plugins.sh.in]: I don't know if you care (not sure which platforms you're targeting), but on macOS the equivalent env var is `DYLD_LIBRARY_PATH` and on Windows it's just the `PATH` that needs to be set for `.dll`'s to be found.

Comment (wjwwood) [reply gazebo_plugins/env-hooks/gazebo_plugins.sh.in]: The `PATH` is updated already (so nothing new needs to be done there), but here's an example of us setting things conditionally with the library path:

https://github.com/ament/ament_package/blob/master/ament_package/template/environment_hook/library_path.sh

Comment (chapulina) [reply gazebo_plugins/env-hooks/gazebo_plugins.sh.in]: Thanks, we shouldn't be breaking Mac or Windows, even though there's no CI running for them. Implemented your suggestion on 55303ac832b7ebd6a68243d5d53562e4752185fb and verified it doesn't break Ubuntu :grin:

**Review by wjwwood — APPROVED** [review]

**Comment by traversaro** [comment]
Just as a cross-ref for anyone experiencing the same issue, I am not sure this fix is working on recent versions of macOS due to `DYLD_LIBRARY_PATH` sanitification, see https://github.com/RoboStack/ros-noetic/issues/462#issuecomment-2041525606 .

**Pull request merged**